src/app/[locale]/specs/sync/page.mdx

```diff
@@ -0,0 +1,173 @@
+- mirroring services, which retain a full copy of repository data, should verify that commit diffs leave the MST tree in a complete and valid state (eg, no missing records, no invalid MST nodes, commit CID would be reproducible if the MST structure was re-generated from scratch)
+- Relays (specifically) should not validate records against Lexicons
+
+## Reliable Synchronization
+
+This section describes some details on how to reliably subscribe to the firehose and maintain an existing synchronized mirror of the network.
+
+Services should generally maintain a few pieces of state for all accounts they are tracking data from:
+
+- track the most recent commit `rev` they have successfully processed
+- keep cached identity data, and use cache expiration to ensure periodic re-validation of that data
+- track account status
+
+Identity caches should be purged any time an `#identity` event is received. Additionally, identity resolution should be refreshed if a commit signature fails to verify, in case the signing key was updated but the identity cache has not been updated yet.
+
+When `tooBig` events are emitted on the firehose, downstream services will need to fetch the diff out-of-band. This usually means an API request to the `com.atproto.sync.getRepo` endpoint on the current PDS host for the account, with the `since` field included. The `since` value should be the most recently processed `rev` value for the account, which may or may not match the `since` field in the commit event message.
+
+If a `#commit` is received with a `since` that does not match the most recently processed `rev` for the account, and is “later” (higher value) than the most recent commit `rev` the service has processed for that account, the service may need to do the same kind of out-of-band fetch as for a `tooBig` event.
+
+Services should keep track of the `seq` number of their upstream subscriptions. This should be stored separately per-upstream, even if there is only a single Relay connection, in case a different Relay is subscribed to in the future (which will have different `seq` numbers).
+
+Events can be processed concurrently, but they should be processed sequentially in-order for any given account. This can be accomplished by partitioning multiple workers using the repo DID as a partitioning key.
+
+Services can confirm that they are consuming content reliably by fetching a snapshot of repository DIDs and `rev` numbers from other services, including PDS hosts and Relay instances. After a short delay, these can be compared against the current state of the service to identify any accounts which have lower than expected `rev` numbers. These repos can then be updated out-of-band.
```

Comment:
How? Would this mechanism be off-protocol? Or `listRepos`? ...which is doable, but wouldn't be a snapshot, right?